### PR TITLE
Remove source/destination spec IDs and replace with source/destinationIds in the backend

### PR DESCRIPTION
**Pull request by sherifnada**
## What
This PR removes spec IDs as first class citizens from everywhere in the (backend) codebase.

### CODE REVIEW

**Review by jrhizor — APPROVED** [review]

File: airbyte-api/src/main/openapi/config.yaml [hunk 1]
```diff
@@ -895,21 +925,15 @@ components:
...
     SourceSpecification:
       description: The specification for what values are required to configure the source.
       example: { user: { type: string } }
     SourceSpecificationRead:
       type: object
       required:
-        - sourceSpecificationId
         - sourceId
-        - specification
+        - connectionSpecification
```
Comment (michel-tricot) [inline airbyte-api/src/main/openapi/config.yaml:935]: the naming is confusing here and it introduces a new term `connection`
what about something like: `expectedConfigurations` or `configurations`

File: airbyte-api/src/main/openapi/config.yaml [hunk 2]
```diff
@@ -875,16 +875,46 @@ components:
       properties:
         sourceId:
           $ref: "#/components/schemas/SourceId"
+    SourceCreate:
```
Comment (michel-tricot) [inline airbyte-api/src/main/openapi/config.yaml:878]: Shouldn't that be called SourceDefinitionCreate based on https://airbytehq.slack.com/archives/C019WEENQRM/p1601672342057100

File: airbyte-api/src/main/openapi/config.yaml [hunk 3]
```diff
@@ -875,16 +875,46 @@ components:
...
+      description: Update the source. Currently, the only allowed attribute to update is the default docker image version.
+      required:
+        - sourceId
+        - dockerImageTag
+      properties:
+        sourceId:
+          $ref: "#/components/schemas/SourceId"
+        dockerImageTag:
+          type: string
     SourceRead:
```
Comment (michel-tricot) [inline airbyte-api/src/main/openapi/config.yaml:902]: SourceDefinitionRead

**Review by michel-tricot — COMMENTED** [review]
I am very confused with the models naming (doesn't seem to match what we discussed in https://airbytehq.slack.com/archives/C019WEENQRM/p1601672342057100).

**Comment by sherifnada** [comment]
@michel-tricot gonna do the rename in a future PR. I felt the scope of these changes was already decently large. That ok with you?

**Review by michel-tricot — APPROVED** [review]

**Pull request merged**